openhtf/core/test_state.py

```diff
@@ -130,7 +146,53 @@ def test_api(self):
                 running_phase_state.attachments,
                 running_phase_state.attach,
                 running_phase_state.attach_from_file,
-                self.notify_update))
+                self.get_measurement,
+                self.get_attachment,
+                self.notify_update,))
+
+  def get_measurement(self, measurement_name):
+    """Get a copy of a measurement value from current or previous phase.
+
+    Args:
+      measurement_name: str of the measurement name
+    Returns:
+      an ImmutableMeasurement
+    """
+    # The framework ignores measurements from SKIP and REPEAT phases
+    ignore_outcomes = set([test_record.PhaseOutcome.SKIP,])
+
+    # Check current running phase state
+    if self.running_phase_state:
+      if measurement_name in self.running_phase_state.measurements:
+        return ImmutableMeasurement.FromMeasurement(
+            self.running_phase_state.measurements[measurement_name])
+
+    # Iterate through phases in reversed order for LIFO.
+    for phase_record in reversed(self.test_record.phases):
+      if (phase_record.result not in ignore_outcomes and
+          measurement_name in phase_record.measurements):
+        measurement = phase_record.measurements[measurement_name]
+        return ImmutableMeasurement.FromMeasurement(measurement)
+
+    test.logger.warning('Could not find measurement: %s', measurement_name)
+    return None
```

Comment:
Since measurements can be `None`, I don't think it's a good return value to use to mean "not found." In my [mockup](https://github.com/google/openhtf/issues/623#issuecomment-319538471) of this functionality I've used `LookupError` to signal that nothing valid is found.

Reply:
This returns an ImmutableMeasurement which.  Even if a measurement value is set at None, it will show up as as the value attribute.  

```
test.measurements.my_measure = None
my_measure = test.get_measurement('my_measure')

my_measure is None # False
my_measure.value is None # True

other_measurement = test.get_measurement('other_measurement')
other_measurement is None # True